### PR TITLE
Nebraska Military Retirement Benefit Exclusion 2022

### LINKED ISSUE #5011: Nebraska Military Retirement Benefit Exclusion 2022
NE military retirement income

[Reference](https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/2022_Ne_Individual_Income_Tax_Booklet_8-307-2022_final_8.pdf)

### PR DESCRIPTION
**Pull request by pxu12**
Fixes #5011

### CODE REVIEW

File: policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py:1]: we dont need to cap it at fagi here - we will do it later on in the state AGI

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py:1]: lets add it to the list of subtractions in ne

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+description: Fraction of military retirement benefits allowed as NE AGI subtraction.
+values:
+  2022-01-01: 1
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:3]: could we add 2021 value as well?

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+metadata:
+  label: fraction of military retirement benefits allowed as NE AGI subtraction
+  unit: /1
+  period: year
+  reference:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:9]: add 2021 and 2023 tax form please

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+description: Fraction of military retirement benefits allowed as NE AGI subtraction.
+values:
+  2022-01-01: 1
+
+metadata:
+  label: fraction of military retirement benefits allowed as NE AGI subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:6]:
```suggestion
  label: Nebraska military retirement subtraction fraction
```

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+description: Fraction of military retirement benefits allowed as NE AGI subtraction.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:1]:
```suggestion
description: Nebraska subtracts this fraction of military retirement benefits from federal adjusted gross income.
```

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  2022-01-01: 1
+
+metadata:
+  label: fraction of military retirement benefits allowed as NE AGI subtraction
+  unit: /1
+  period: year
+  reference:
+  - title: 2022 NE income tax form and instruction booklet
+    href: https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/2022_Ne_Individual_Income_Tax_Booklet_8-307-2022_final_8.pdf#page=3
+  - title: Legislative Bill 387
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:12]: could we add the actual legal code section? - The section that is amended by this bill

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml (hunk 1 above)
Comment (pxu12) [reply policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:4]: Screenshot 2024-09-12 at 11 08 09
On the 2021 tax form, the percentage is 40%. However, in the legal code, from 2015 to 2021, the tax filer could choose between receiving 40% for 7 years or 15% for all years after age 67. I'm not sure how to represent this. 
Screenshot 2024-09-12 at 11 09 38

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:4]: Lets represent this as the 40% described in the tax form and add a note in the variable

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  reference:
+  - title: 2021 Nebraska Individual Income Tax Booklet
+    href: https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/f_1040n_booklet.pdf#page=16
+  - title: 2022 NE income tax form and instruction booklet
+    href: https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/2022_Ne_Individual_Income_Tax_Booklet_8-307-2022_final_8.pdf#page=3
+  - title: 2023 Nebraska Individual Income Tax Booklet
+    href: https://revenue.nebraska.gov/sites/revenue.nebraska.gov/files/doc/tax-forms/2023/incometax/f_1040n_booklet_2023_Final.pdf#page=19
+  - title: Nebraska Revised Statute 77-2716(15)
+    href: https://nebraskalegislature.gov/laws/statutes.php?statute=77-2716
+  - title: Legislative Bill 387
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/military_retirement/fraction.yaml:19]: we can remove the bill here

File: policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml [hunk 7]
```diff
@@ -5,6 +5,7 @@ values:
   2024-01-01:
     - ne_social_security_subtraction
     - taxable_public_pension_income
+    - ne_military_retirement_subtraction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ne/tax/income/agi/subtractions/subtractions.yaml:8]: also add in 2021

File: policyengine_us/tests/policy/baseline/gov/states/ne/tax/income/adjusted_gross_income/ne_military_retirement_subtraction.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+- name: Test 1, less than agi, for 2022
+  absolute_error_margin: 0.01
+  period: 2022
+  input:
+     military_retirement_pay: 30_000
+     state_code: NE
+  output:
+    ne_military_retirement_subtraction: 30_000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/ne/tax/income/adjusted_gross_income/ne_military_retirement_subtraction.yaml:8]: add 2021 test

File: policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+from policyengine_us.model_api import *
+
+
+class ne_military_retirement_subtraction(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py:4]: looking at the [law](https://nebraskalegislature.gov/laws/statutes.php?statute=77-2716):
I think the person has to be at least 67

Could you add this condition?

image

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.ne.tax.income.agi.subtractions.military_retirement
+        person = tax_unit.members
+        age = person("age", period)
+        age_eligible = age >= p.age_threshold
+        military_retirement_benefit = person("military_retirement_pay", period)
+        military_retirement_benefits = tax_unit.sum(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ne/tax/income/adjusted_gross_income/subtractions/ne_military_retirement_substraction.py:20]:
```suggestion
        qualifying_military_retirement_benefits = tax_unit.sum(
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**